Firmware/LowLevel/src/datatypes.h

```diff
@@ -135,22 +137,62 @@ struct ll_ui_event {
 } __attribute__((packed));
 #pragma pack(pop)
 
-#define LL_HIGH_LEVEL_CONFIG_MAX_COMMS_VERSION 1           // Max. comms packet version supported by this open_mower LL FW
 #define LL_HIGH_LEVEL_CONFIG_BIT_DFPIS5V 1 << 0            // Enable full sound via mower_config env var "OM_DFP_IS_5V"
-#define LL_HIGH_LEVEL_CONFIG_BIT_EMERGENCY_INVERSE 1 << 1  // Sample, for possible future usage, i.e. for SA-Type emergency
+#define LL_HIGH_LEVEL_CONFIG_BIT_BACKGROUND_SOUNDS 1 << 1  // Enable background sounds
 
 typedef char iso639_1[2]; // Two char ISO 639-1 language code
 
+enum class HallMode : uint8_t {
+    off = 0,
+    emergency,  // lift & tilt
+    stop,
+    pause
```

Comment:
> I don't really know the implications of these modes, mind to explain (and/or add a comment)? I assume it has some to do with which of the timeouts to use?

Yes, with both. Commented, but describing "lift" and "tilt" differences would be the wrong place there. That is, or should be done where the emergency get handled.

> Also, the enum values are basically constants, so I'd write them in upper-case.

I'm always unsure with this. If no style guide exists, I normally **try** follow "Google C++ Style Guide", but by enums i.e., they write that they should be named like const but NOT macros https://google.github.io/styleguide/cppguide.html#Enumerator_Names !
I never would write enum class member in UPPERCASE like macros but I also do NOT like google "k" prefix naming rule. Often tried, but looks and writes always ugly :-/

Reply:
I looked at the same guide... and agree that the "k" prefix looks strange. Not sure what else I'd use for real constants... I think upper-case. Oh right, [I did]( https://github.com/Apehaenger/OpenMower/commit/1c0cfacb801e5e21bd3c39f4b6be79a5c9e25708#diff-91326d111ce913c238d2694e904666d67eea719cc8f00bfb954e9f58ef421db6R122). So I guess for me, macros and constants share one style. And since the enum is basically just a namespace/prefix, I truly think that `HallLevel::ACTIVE_LOW` looks better than `HallLevel::activeLow`. Arduino seems to agree:
```c++
typedef enum {
  INPUT           = 0x0,
  OUTPUT          = 0x1,
  INPUT_PULLUP    = 0x2,
  INPUT_PULLDOWN  = 0x3,
  OUTPUT_2MA      = 0x4,
  OUTPUT_4MA      = 0x5,
  OUTPUT_8MA      = 0x6,
  OUTPUT_12MA     = 0x7,
} PinMode;
```

**But**: I already talked you out of so much of your original code already, and this is really an unimportant detail, so do what you like, I won't complain if you keep your style. 😆

Reply:
I'm fully fine with your hint!
I don't have community-wise dev experience, and I really like having all that rules. But I miss that routine, I still need to think how was what written?

Regarding the enum: You're right! Clemens also write them in UPPERCASE. In the same file! So, for sure I follow that style.